src/lib/spinel/radio_spinel.hpp

```diff
@@ -998,7 +998,8 @@ template <typename InterfaceType, typename ProcessContextType> class RadioSpinel
                                         spinel_prop_key_t aKey,
                                         const char       *aFormat,
                                         va_list           aArgs);
-    otError WaitResponse(void);
+    otError WaitResponse(bool aWaitingReset = false);
+    otError WaitForResetReason(void) { return WaitResponse(true); }
```

Comment:
I suggest not adding a dedicated method to wait for Reset Reason and simply call `WaitResponse` for the reason I mentioned in another comment.